website/server/controllers/api-v3/tasks.js

```diff
@@ -16,6 +16,27 @@ import logger from '../../libs/api-v3/logger';
 
 let api = {};
 
+async function _checkShortNameUniqueness (user, tasks) {
+  let shortNames = tasks.map(task => task.shortName).filter(name => name);
+
+  if (shortNames.length !== [...new Set(shortNames)].length) {
```

Comment:
Yeah, the local data checks against creating multiple tasks with the same shortName. There's no way to catch that with a mongo query, since the tasks haven't been saved yet.

I'll leave a comment about Set.